src/main.rs

```diff
@@ -151,7 +133,7 @@ fn main() -> Result<(), aes_gcm_siv::Error> {
         let mut hasher = Sha256::new();
         hasher.update(x_i.to_le_bytes());
         let result = hasher.finalize();
-        let h_1_x_i = <Fq as PrimeField>::from_le_bytes_mod_order(result[..8].try_into().unwrap());
+        let h_1_x_i = <Fq as PrimeField>::from_le_bytes_mod_order(result[..].try_into().unwrap());
```

Comment:
i see we repeat this conversion often. may be worth pulling this to method. since its a conversion could just implement the From or Into traits